test/support/query.rb

```diff
@@ -0,0 +1,20 @@
+# Test helpers to deal with query strings.
+
+def parse_query(query)
+  if query.nil?
+    {}
+  else
+    query_options query
+  end
+end
+
+def query_options(query)
+  options = {}
+
+  query.split('&').each do |pair|
+    k, v = pair.split('=')
+    options[k.to_sym] = v
```

Comment:
The return format of `CGI::parse` makes the code for this much uglier in `test/support/response.rb`:

```ruby
index_response factory_name, options.fetch('count', [5]).first.to_i
```

rather than the current:

```ruby
index_response factory_name, options.fetch('count', 5).to_i
```

Thoughts?